src/parser/AstNode.ts

```diff
@@ -0,0 +1,28 @@
+import { Location } from "../lexer";
+
+/**
+ * ABC (that's minimally compliant with ESLint/ESTree) for all nodes within
+ * the AST generated by our parser.
+ */
+export abstract class AstNode {
+    _type!: string;
```

Comment:
Any particular reason to call this `_type`?  I don't have a problem with it, I'm just curious since it doesn't seem to be required by ESTree, and the `type()` getter seems like it'll be used for everything anyway 😄

Reply:
It's because I don't know how to Typescript/Javascript :p I'll get rid of the accessor and change this to a readonly member called `type`. I think I was just fumbling over things when I wrote this earlier. Thanks for pointing it out!